django/contrib/admin/checks.py

```diff
@@ -109,6 +110,52 @@ def _check_raw_id_fields_item(self, obj, model, field_name, label):
             else:
                 return []
 
+    def _check_autocomplete_fields(self, obj):
+        """
+        Check that `autocomplete_fields` only contains field names that are
+        listed on the model.
+        """
+        if not isinstance(obj.autocomplete_fields, (list, tuple)):
+            return must_be('a list or tuple', option='autocomplete_fields', obj=obj, id='admin.E001')
+        else:
+            return list(chain(*[
+                self._check_autocomplete_fields_item(obj, obj.model, field_name, 'autocomplete_fields[%d]' % index)
+                for index, field_name in enumerate(obj.get_autocomplete_fields())
+            ]))
+
+    def _check_autocomplete_fields_item(self, obj, model, field_name, label):
+        """
+        Check an item of `autocomplete_fields`, i.e. check that field named
+        `field_name` exists in model `model` and is a ForeignKey or a
+        ManyToManyField.
+        """
+        try:
+            field = model._meta.get_field(field_name)
+        except FieldDoesNotExist:
+            return refer_to_missing_field(field=field_name, option=label, model=model, obj=obj, id='admin.E002')
+        else:
+            if not field.many_to_many and not isinstance(field, models.ForeignKey):
+                return must_be(
+                    'a foreign key or a many-to-many field',
+                    option=label, obj=obj, id='admin.E003'
+                )
+            related_admin = obj.admin_site._registry.get(field.remote_field.model)
+            if related_admin is None:
+                msg = ('An admin for model "%s" has to be registered'
```

Comment:
I'd follow the indentation pattern of the rest of this page, e.g.:
```
return [
    checks.Error(
        'An admin for model "%s" has to be registered to be'
        'referenced by %s.autocomplete_fields.' % (
            field.remote_field.model.__name__,
            type(obj).__name__,
        ),
        obj=obj.__class__,
        id='admin.E036',
    )
]
```
Also, E036/37 are missing from `docs/ref/checks.txt`.